arbor/morph/segment_tree.cpp

```diff
@@ -87,5 +89,123 @@ ARB_ARBOR_API std::ostream& operator<<(std::ostream& o, const segment_tree& m) {
              << io::sepval(tstr, ' ') <<  "))";
 }
 
+
+// Utilities.
+
+ARB_ARBOR_API segment_tree prune_tag(const segment_tree& in, int tag) {
```

Comment:
This might be simpler to implement based on finding the roots
and then using `split_at(tree, root)` from #1957 or at least the 
utilities introduced there.

Reply:
I've done that in 0695a75, where `split_at` needed a fix to make it work with the L5PC model of BluePyOpt. At this particular commit the L5PC segment trees obtained with `tag_roots`/`split_at` and `prune_tag` for the axon-tag are equivalent. For this reason, `prune_tag` was removed in the follow-up commit 38ec4f7.